src/com/estafet/learning/US006/Task6.java

```diff
@@ -0,0 +1,166 @@
+
+        System.out.println("\n------------------------");
+        System.out.println("Array length is: " + stringArr.length);
+        for (int i = 0; i <= 9; i++) {
+            System.out.print(stringArr[i] + " ");
+        }
+        String first = stringArr[0];
+        String last = stringArr[stringArr.length - 1];
+        System.out.print("\nFirst element of array is : " + first);
+        System.out.print("\nLast element of array is : " + last);
+        System.out.println("\n------------------------");
+    }
+
+    private static void getAverageOfArray() {
+        int[] intArr = new int[3];
+        intArr[0] = 11;
+        intArr[1] = 66;
+        intArr[2] = 5;
+
+        double total = 0;
+        for (int i : intArr) {
+            total = total + i;
+        }
+
+        System.out.println("\nGet Average of Array:");
+        System.out.format("The total is: " + total + "%n");
+        double average = total / intArr.length;
+        System.out.format("The average is: " + average + "%n");
+        System.out.println("\n------------------------");
+    }
+
+    private static void copyElementsOfArray() {
+        int[] a = {1, 4, 8};
+        int[] b;
+        b = Arrays.copyOf(a, 3);
+
+        System.out.format("Elements of Array a are: %n");
+        for (int j : a) System.out.print(j + " ");
+
+        System.out.format("%nElements of Array b are:  %n");
+        for (int j : b) System.out.print(j + " ");
+        System.out.println("\n------------------------");
+    }
+
+    private static void getMinAndMaxOfArray() {
+        int[] a = {13, 43243, 8, 321321312, 43423, -1};
+        Arrays.sort(a);
+        System.out.print("\nMin value of Array is : " + a[0]);
+        System.out.print("\nMax value of Array is : " + a[a.length - 1]);
+    }
+
+    public static int[] addElement(int n, int[] arr, int x, int pos) {
+        int[] newArray = new int[n + 1];
+        for (int i = 0; i < n + 1; i++) {
+            if (i < pos - 1) newArray[i] = arr[i];
+            else if (i == pos - 1) newArray[i] = x;
+            else newArray[i] = arr[i - 1];
+        }
+        return newArray;
+    }
+
+    private static void addValueInArray() {
+        int n = 10;
+        int x = 50;
+
+        int[] arr = {1, 2, 3, 4, 5, 6, 7, 8, 9, 10};
+        System.out.println("\n------------------------");
+        System.out.println("\nInitial Array:\n" + Arrays.toString(arr));
+
+        arr = addElement(n, arr, x, 5);
+        System.out.println("\nArray with " + x + " added:\n" + Arrays.toString(arr));
+        System.out.println("\n------------------------");
+    }
+
+    private static void findDuplicateValuesOfArray() {
+        int[] arr = new int[]{5, 1, 2, 3, 4, 2, 7, 8, 8, 3, 5};
+        System.out.println("\n------------------------");
+        System.out.println("\nDuplicate elements in the given array: ");
+        for (int i = 0; i < arr.length; i++) {
+            for (int j = i + 1; j < arr.length; j++) {
```

Comment:
Optimization problem: what if array will contain a million of elements - will that work?

Reply:
changed.